src/server/editorServices.ts

```diff
@@ -192,6 +192,10 @@ namespace ts.server {
             return this.roots.map(root => root.fileName);
         }
 
+        getScriptKind() {

```

Comment:
here we should ask the editor about the kind. this way we can support in memory buffers that does not have a name or a file on disk